### PR TITLE
Add a workflow and Makefile target to test old GMT versions every Tuesday

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

To test PyGMT with old GMT versions, a new Makefile target and a new workflow are added in this PR.

### Makefile target `test_no_images`

Because the baseline images usually have slight changes for different GMT versions, it makes little sense to compare the baseline images generated by GMT 6.4 when running the tests with GMT 6.3. However, we still want to run these image-building tests to make sure they don't crash or raise errors. This is done by NOT adding the `--mpl` option to pytest.

### New workflow "GMT Backwards Tests"

This workflow is modified from the "GMT Dev Tests" workflow. It will run when a PR is marked ready for review and will run every Tuesday.

Fixes https://github.com/GenericMappingTools/pygmt/issues/1991.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: .github/workflows/ci_tests_backward.yaml [hunk 1]
```diff
@@ -0,0 +1,105 @@
+# This workflow installs PyGMT and runs tests
+
+name: GMT Backward Tests
+
+on:
+  # push:
+  #   branches: [ main ]
+  pull_request:
+    types: [ready_for_review]
```
Comment (seisman) [inline .github/workflows/ci_tests_backward.yaml:9]:
```suggestion
    # types: [ready_for_review]
```

File: .github/workflows/ci_tests_backward.yaml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+          mamba-version: "*"
+          use-mamba: true
+
+      # Install GMT and other required dependencies from conda-forge
+      - name: Install dependencies
+        run: |
+          mamba install gmt=${{ matrix.gmt_version }} numpy \
+                        pandas xarray netCDF4 packaging geopandas \
+                        build dvc make pytest>=6.0 \
+                        pytest-cov pytest-doctestplus pytest-mpl sphinx-gallery tomli
```
Comment (seisman) [inline .github/workflows/ci_tests_backward.yaml:69]:
```suggestion
                        pytest-cov pytest-doctestplus pytest-mpl sphinx-gallery
```

File: .github/workflows/ci_tests_backward.yaml [hunk 3]
```diff
@@ -0,0 +1,105 @@
+# This workflow installs PyGMT and runs tests
+
+name: GMT Backward Tests
+
+on:
+  # push:
+  #   branches: [ main ]
+  pull_request:
+    # types: [ready_for_review]
```
Comment (seisman) [inline .github/workflows/ci_tests_backward.yaml:9]:
```suggestion
    types: [ready_for_review]
```

File: .github/workflows/ci_tests_backward.yaml [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+          mamba-version: "*"
+          use-mamba: true
+
+      # Install GMT and other required dependencies from conda-forge
+      - name: Install dependencies
+        run: |
+          mamba install gmt=${{ matrix.gmt_version }} numpy \
+                        pandas xarray netCDF4 packaging geopandas \
+                        build dvc make pytest>=6.0 \
+                        pytest-cov pytest-doctestplus pytest-mpl sphinx-gallery
```
Comment (seisman) [inline .github/workflows/ci_tests_backward.yaml:69]:
```suggestion
                        pytest-cov sphinx-gallery
```

File: .github/workflows/ci_tests_backward.yaml [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+          mv .gmt/* ~/.gmt
+          # Change modification times of the two files, so GMT won't refresh it
+          touch ~/.gmt/server/gmt_data_server.txt ~/.gmt/server/gmt_hash_server.txt
+          ls -lhR ~/.gmt
+
+      # Pull baseline image data from dvc remote (DAGsHub)
+      - name: Pull baseline image data from dvc remote
+        run: |
+          dvc pull --verbose
+          ls -lhR pygmt/tests/baseline/
```
Comment (seisman) [inline .github/workflows/ci_tests_backward.yaml:97]:
```suggestion
```

File: .github/workflows/ci_tests_backward.yaml [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+          miniforge-variant: Mambaforge
+          mamba-version: "*"
+          use-mamba: true
+
+      # Install GMT and other required dependencies from conda-forge
+      - name: Install dependencies
+        run: |
+          mamba install gmt=${{ matrix.gmt_version }} numpy \
+                        pandas xarray netCDF4 packaging geopandas \
+                        build dvc make pytest>=6.0 \
```
Comment (seisman) [inline .github/workflows/ci_tests_backward.yaml:68]:
```suggestion
                        build make pytest>=6.0 \
```

File: .github/workflows/ci_tests_backward.yaml [hunk 7]
```diff
@@ -0,0 +1,105 @@
+# This workflow installs PyGMT and runs tests
```
Comment (seisman) [inline .github/workflows/ci_tests_backward.yaml:1]:
```suggestion
# This workflow installs PyGMT and runs tests with old GMT versions
```

File: .github/workflows/ci_tests_backward.yaml (hunk 5 above)
Comment (maxrjones) [inline .github/workflows/ci_tests_backward.yaml:97]: Is pulling the baseline images needed even without ``--mpl``?

Comment (seisman) [reply .github/workflows/ci_tests_backward.yaml:97]: Baseline images are not needed for all tests except one (https://github.com/GenericMappingTools/pygmt/blob/main/pygmt/tests/test_image.py). This test needs to read the `test_logo.png` image in the baseline directory.

File: .github/workflows/ci_tests_backward.yaml [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+    paths-ignore:
+      - 'doc/**'
+      - 'examples/**'
+      - '*.md'
+      - 'README.rst'
+      - 'LICENSE.txt'
+      - '.gitignore'
+  # Schedule tests on Monday/Wednesday/Friday
+  schedule:
+    - cron: '0 0 * * 1,3,5'
```
Comment (weiji14) [inline .github/workflows/ci_tests_backward.yaml:19]: Do we need to test so frequently for GMT 6.3? Maybe just Tuesday and Thursday?

Comment (seisman) [reply .github/workflows/ci_tests_backward.yaml:19]: I'm OK with a less frequent CI job for old GMT versions.

At line 9, perhaps we also don't need to run the CI job for when a PR is marked as ready for review.

File: .github/workflows/ci_tests_backward.yaml [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+          mkdir -p ~/.gmt
+          mv .gmt/* ~/.gmt
+          # Change modification times of the two files, so GMT won't refresh it
+          touch ~/.gmt/server/gmt_data_server.txt ~/.gmt/server/gmt_hash_server.txt
+          ls -lhR ~/.gmt
+
+      # Pull baseline image data from dvc remote (DAGsHub)
+      - name: Pull baseline image data from dvc remote
+        run: |
+          dvc pull --verbose
```
Comment (maxrjones) [inline .github/workflows/ci_tests_backward.yaml:96]:
```suggestion
          dvc pull pygmt/tests/baseline/test_image.png --verbose
```
The dvc pull step is a common failure point, so best to only pull the one file needed.

File: .github/workflows/ci_tests_backward.yaml [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+  pull_request:
+    # types: [ready_for_review]
+    paths-ignore:
+      - 'doc/**'
+      - 'examples/**'
+      - '*.md'
+      - 'README.rst'
+      - 'LICENSE.txt'
+      - '.gitignore'
+  # Schedule tests on Monday/Wednesday/Friday
```
Comment (seisman) [inline .github/workflows/ci_tests_backward.yaml:17]:
```suggestion
  # Schedule tests on Tuesday
```

File: .github/workflows/ci_tests_backward.yaml (hunk 8 above)
Comment (seisman) [inline .github/workflows/ci_tests_backward.yaml:19]:
```suggestion
    - cron: '0 0 * * 2'
```

Comment (seisman) [reply .github/workflows/ci_tests_backward.yaml]: > Do we need to test so frequently for GMT 6.3? Maybe just Tuesday and Thursday?

Updated in 8e729ebb950506d00ac50c4b660ddb0a71d8e911.

File: .github/workflows/ci_tests_backward.yaml [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+          mkdir -p ~/.gmt
+          mv .gmt/* ~/.gmt
+          # Change modification times of the two files, so GMT won't refresh it
+          touch ~/.gmt/server/gmt_data_server.txt ~/.gmt/server/gmt_hash_server.txt
+          ls -lhR ~/.gmt
+
+      # Pull baseline image data from dvc remote (DAGsHub)
+      - name: Pull baseline image data from dvc remote
+        run: |
+          dvc pull pygmt/tests/baseline/test_image.png --verbose
```
Comment (seisman) [inline .github/workflows/ci_tests_backward.yaml:96]:
```suggestion
          dvc pull pygmt/tests/baseline/test_logo.png --verbose
```

**Comment by seisman** [comment]
Ping @weiji14 and @maxrjones for another round of review.

File: .github/workflows/ci_tests_backward.yaml [hunk 12]
```diff
@@ -0,0 +1,105 @@
+# This workflow installs PyGMT and runs tests with old GMT versions
+
+name: GMT Backward Tests
```
Comment (weiji14) [inline .github/workflows/ci_tests_backward.yaml:3]: Not so sure if 'backward' is a good word, how about 'legacy' (https://en.wikipedia.org/wiki/Legacy_system)? Maybe we need to ask a native English speaker :slightly_smiling_face:

```suggestion
name: GMT Legacy Tests
```

**Review by weiji14 — COMMENTED** [review]
Some suggestions to run these tests on lower versions of Python and Operating Systems.

File: .github/workflows/ci_tests_backward.yaml (hunk 12 above)
Comment (seisman) [reply .github/workflows/ci_tests_backward.yaml:3]: *Legacy* sounds good and let's wait for some inputs for native English speakers.

File: .github/workflows/ci_tests_backward.yaml [hunk 13]
```diff
@@ -0,0 +1,105 @@
+# This workflow installs PyGMT and runs tests with old GMT versions
```
Comment (seisman) [inline .github/workflows/ci_tests_backward.yaml:1]: Remember to rename the workflow file if we decide to call it "GMT Legacy Tests"

File: .github/workflows/ci_tests_backward.yaml (hunk 12 above)
Comment (seisman) [reply .github/workflows/ci_tests_backward.yaml]: Done in be81f365240de56636f4d3d324a3417f00090e9c

File: .github/workflows/ci_tests_backward.yaml (hunk 13 above)
Comment (seisman) [reply .github/workflows/ci_tests_backward.yaml]: Done in be81f365240de56636f4d3d324a3417f00090e9c.

File: .github/workflows/ci_tests_legacy.yaml [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+  #   branches: [ main ]
+  pull_request:
+    # types: [ready_for_review]
+    paths-ignore:
+      - 'doc/**'
+      - 'examples/**'
+      - '*.md'
+      - 'README.rst'
+      - 'LICENSE.txt'
+      - '.gitignore'
```
Comment (weiji14) [inline .github/workflows/ci_tests_legacy.yaml:16]: Comment these out before merging.

**Review by weiji14 — APPROVED** [review]

**Review by willschlitzer — APPROVED** [review]

File: .github/workflows/ci_tests_legacy.yaml (hunk 14 above)
Comment (seisman) [inline .github/workflows/ci_tests_legacy.yaml:16]:
```suggestion
  # pull_request:
    # types: [ready_for_review]
    # paths-ignore:
    #  - 'doc/**'
    #  - 'examples/**'
    #  - '*.md'
    #  - 'README.rst'
    #  - 'LICENSE.txt'
    #  - '.gitignore'
```

**Pull request merged**